functionary/builder/tests/test_utils.py

```diff
@@ -119,3 +125,37 @@ def test_delete_removed_function_parameters(function1):
 
     assert function1.parameters.count() == 1
     assert function1.parameters.filter(name="param1").exists()
+
+
+def test_invalid_package_yaml(mocker, package_contents):
```

Comment:
These tests are a little bit in the weeds in that they feel like they're testing specific implementation details.  I pushed up a test in `views/test_publish.py` that is more of what I had in mind when I suggested adding tests.  That is, it reproduces the actual bug from the API user perspective: malformed yaml is provided during the publish and the server throws an error rather responding with a 400.  That test blows up if run against the code prior to your changes.  With your changes, the test passes.  When addressing bugs, I think that's generally the style of testing that's most useful because it specifically confirms that the the bug has been addressed, as opposed to confirming that some internal implementation detail has improved in a way that only sort of implies the bug has been fixed.

Reply:
Gotcha. I'll start including integration and regression tests with my unit tests